test/fuzz/lua_fuzzer/lua_grammar.proto

```diff
@@ -0,0 +1,310 @@
+    }
+}
+
+message DoBlock {
+    required LuaBlock block = 1;
+}
+
+message WhileCycle {
+    required Expression condition = 1;
+    required LuaBlock code = 2;
+}
+
+message RepeatCycle {
+    required LuaBlock code = 1;
+    required Expression condition = 2;
+}
+
+message IfStatement {
+    required Expression firstCondition = 1;
+    required LuaBlock firstBlock = 2;
+    repeated ElseIf clauses = 3;
+    optional LuaBlock lastBlock = 4;
+}
+
+message ElseIf {
+    required Expression condition = 1;
+    required LuaBlock code = 2;
+}
+
+message ForCycleName {
+    required string name = 1;
+    required Expression init = 2;
+    required Expression max_min = 3;
+    optional Expression increment = 4;
+    required LuaBlock code = 5;
+}
+
+message ForCycleList {
+    required NameList names = 1;
+    required ExpressionList expressions = 2;
+    required LuaBlock code = 3;
+}
+
+message Function {
+    required FuncName name = 1;
+    required FuncBody body = 2;
+}
+
+message FuncName {
+    required string firstName = 1;
+    repeated string names = 2;
+    optional string lastName = 3;
+}
+
+message FuncBody {
+    optional ParList list = 1;
+    required LuaBlock block = 2;
+}
+
+message ParList {
+    oneof parlist_oneof {
+        NameListWithEllipsis list = 1;
+        string ellipsis = 2;
+    }
+}
+
+message NameListWithEllipsis {
+    required NameList names = 1;
+    optional string ellipsis = 2;
+}
+
+message NameList {
+    required string firstName = 1;
+    repeated string names = 2;
+}
+
+message LocalFunc {
+    required string name = 1;
+    required FuncBody body = 2;
+}
+
+message LocalNames {
+    required NameList names = 1;
+    optional ExpressionList expressions = 2;
+}
+
+message LastStatement {
+    oneof last_oneof {
+        ReturnLast return = 1;
+        uint32 break = 2;
+    }
+}
+
+message ReturnLast {
+    optional ExpressionList exprList = 1;
+}
+
+message VariableList {
+    required Variable var = 1;
+    repeated Variable varNext = 2;
+}
+
+message ExpressionList {
+    repeated Expression expr = 1;
+    required Expression lastExpr = 2;
+}
+
+message OptionalExpressionList {
+    optional ExpressionList list = 1;
+}
+
+message Variable {
+    oneof var_oneof {
+        string name = 1;
+        PrefixExpressionWithExpression expr = 2;
+        PrefixExpressionWithName exprName = 3;
+    }
+}
+
+message PrefixExpressionWithExpression {
+    required PrefixExpression prefix = 1;
+    required Expression expr = 2;
+}
+
+message PrefixExpressionWithName {
+    required PrefixExpression prefix = 1;
+    required string name = 2;
+}
+
+message PrefixExpression {
+    oneof prefix_oneof {
+        Variable var = 1;
+        FunctionCall call = 2;
+        Expression expr = 3;
+    }
+}
+
+message Expression {
+    oneof expr_oneof {
+        uint32 nil = 1;
+        uint32 false = 2;
+        uint32 true = 3;
+        int64 number = 4;
+        string str = 5;
+        uint32 ellipsis = 6;
```

Comment:
Actually, it doesn't really matter, because ellipsis value which is generated by protobuf is never used itself, this field is just used for checking whether the message has ellipsis field. Changed type to 'string' for more unified codebase.